### DISCUSSION TITLE
Question about `QueryCache.config`

### DISCUSSION THREAD

**Discussion by 87xie** [TanStack/query#7576] [Q&A] [answered]
I have a question regarding the `config` parameter in the `QueryCache.prototype.constructor`. In the current implementation, it accepts a config parameter. However, I have not found any instances of `QueryCache` in the codebase that are created with a `config` parameter.

```ts
class QueryCache extends Subscribable<QueryCacheListener> {
  #queries: QueryStore

  constructor(public config: QueryCacheConfig = {}) { // <-- here
    super()
    this.#queries = new Map<string, Query>()
  }
  // ...
}
```

Specifically, I'm interested in understanding if the `config` parameter is currently not in use, would it make sense to remove it, or is it reserved for potential future extensions?

Any clarification or examples of how this `config` parameter is meant to be used would be greatly appreciated!

**Comment by TkDodo** [comment, marked as answer]
it's for adding callbacks: https://tanstack.com/query/v5/docs/reference/QueryCache
